src/challenges.js

```diff
@@ -5,25 +5,56 @@
 const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
+  return elements.forEach(cb);
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function.
   // Return the new array.
+  return elements.map(cb);
+
+  // let myMap = [];
+  // for (let i = 0; i < elements.length; i++) {
+  //   myMap = elements[i].push(myMap);
+  // }
+  // return myMap;
+
 };
 
 /* ======================== Closure Practice ============================ */
 // No test needed here, just run the newCounter(); and make sure it's counting up
 const counter = () => {
   // Return a function that when invoked increments and returns a counter variable.
   // Example: const newCounter = counter();
-  // newCounter(); // 1
+  // newCounter(); // 1   
   // newCounter(); // 2
-};
+  
+  const counter = () => {
+    let num = 0;
+    return function() 
+    { return num++ }
```

Comment:
You should be incrementing before you return `++num` is the easiest way to accomplish this.